### ISSUE TITLE
[Android] UnityAds does not load with Mediation 

### ISSUE THREAD

**Issue by wasedaigo** [invertase/react-native-firebase#1792]
:fire::fire::fire:
## Issue
RewardedVideo of UnityAds does not show up on Android with Admob Mediation

I found the following warning
```
UnityAdapter: Context is not an Activity. 
Unity Ads requires an Activity context to load ads.
```

Changing getContext() into getActivity() fixed the problem, however I do not know if this is the right solution.  (I modified the following line)
https://github.com/invertase/react-native-firebase/blob/master/android/src/main/java/io/invertase/firebase/admob/RNFirebaseAdMobRewardedVideo.java#L28

I assume other mediation adapters such as adcolony, appodeal also have the same issue, as I read somewhere they also need activity context.

## Environment

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [x] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **If known, the version of the platform are you experiencing the issue on:**
  - `Android API 21`
- **Operating System:**
  - [x] MacOS, version: `N/A`
  - [ ] Windows, version: `N/A`
  - [ ] Other, please specify: `N/A`
- **Build Tools:**
  - `Android Studio 3.2`
- **`React Native` 
  - `version:0.57.4`
- **`React Native Firebase` 
  - `library version:5.1.0`
- **`Firebase` module(s) you're using that has the issue:**
  - [x] **N/A**
  - [ ] Authentication
  - [ ] Analytics
  - [ ] Cloud **Firestore**
  - [ ] Cloud **Messaging** (FCM)
  - [ ] Crashlytics
  - [ ] Dynamic **Links**
  - [ ] **Functions** Callable
  - [ ] Invites
  - [ ] Instance ID
  - [ ] Notifications
  - [ ] Performance Monitoring
  - [ ] Realtime **Database**
  - [ ] Remote **Config**
  - [ ] Storage
- **Are you using `TypeScript`?**
  - [x] No
  - [ ] Yes, version: `N/A`
- **Are you using Expo, e.g. `ExpoKit`?**
  - [x] No
  - [ ] Yes, I've _not_ ejected
  - [ ] Yes, but I **have** ejected to `ExpoKit`
  - [ ] Yes, but I **have** ejected to vanilla React Native
  - Expo version: `N/A`

**Comment by wasedaigo** [comment]
No Mr. bot, I made a PR ...
https://github.com/invertase/react-native-firebase/pull/1892

**Comment by Salakar** [comment]
Pinned it 🙈 thanks again for the PR, will take a look once CI is back to normal

**Comment by wasedaigo** [comment]
@Salakar Thank you very much!

**Comment by mikehardy** [comment]
haha - I think we fixed this one!

**Comment by tarouboy** [comment]
Hi @wasedaigo 
it happens on v10.3 again.
Failed to load ad: 105: Unity Ads requires an Activity context to load ads.

Do you know how to fix it? 🆘 🙏

**Comment by wasedaigo** [comment]
I am not using ReactNative anymore so... I don't know!
Maybe refer to this PR and see what is missing in the current codebase?

https://github.com/invertase/react-native-firebase/pull/1892/files

**Comment by zyzo** [comment]
Anyone found out how to fix this ? I'm using the latest version of `@react-native-firebase/admob` and I'm seeing this issue (only for Android): 

package.json
```
  "@react-native-firebase/admob": "^10.7.0",
  "@react-native-firebase/app": "^10.7.0",
  "react-native": "0.63.2",
```

app/build.gradle
```
 implementation 'com.google.android.gms:play-services-ads:19.6.0'
 implementation 'com.google.ads.mediation:applovin:9.14.12.0'
 implementation 'com.google.ads.mediation:unity:3.6.0.0'
```

**Comment by zyzo** [comment]
I was able to display Unity ads with this fix
https://github.com/BetOnYou/react-native-firebase/commit/260ceee045a7af61ebeaec7728f54c28bbd0eaac

**Comment by mikehardy** [comment]
@zyzo if that's something we need to change, I would happily collaborate to merge a PR if you posted one. One thing to pay attention to in react-native is that it is *possible* at any time that the Activity context is unexpectedly null when you ask for it, because the Activity react-native uses can go away before the native request has asynchronously been passed down to native and started executing. But if you just carefully check for null when request the activity context and have a reasonable fallback (and maybe a native info log message for developers so they know what happened when the ad failed to load rarely) then it could be an easy PR

**Comment by tarouboy** [comment]
> @zyzo if that's something we need to change, I would happily collaborate to merge a PR if you posted one. One thing to pay attention to in react-native is that it is _possible_ at any time that the Activity context is unexpectedly null when you ask for it, because the Activity react-native uses can go away before the native request has asynchronously been passed down to native and started executing. But if you just carefully check for null when request the activity context and have a reasonable fallback (and maybe a native info log message for developers so they know what happened when the ad failed to load rarely) then it could be an easy PR

However without the Activity context it won't even load.
`Failed to load ad: 105: Unity Ads requires an Activity context to load ads.`




> I was able to display Unity ads with this fix
> [BetOnYou@260ceee](https://github.com/BetOnYou/react-native-firebase/commit/260ceee045a7af61ebeaec7728f54c28bbd0eaac)

It also applies on Interstitial Ads

**Comment by mikehardy** [comment]
fix Releasing as 10.8.1 right now, thanks!
